enhancements/sig-architecture/64-placementStrategy/README.md

```diff
@@ -0,0 +1,434 @@
+status:
+  decisions:
+    - clusterName: cls006
+      reason: ''
+    ...
+---
+apiVersion: cluster.open-cluster-management.io/v1beta1
+kind: PlacementDecision
+metadata:
+  name: ztp-placement-decision-2
+  namespace: ztp-acm-ns
+  labels:
+    cluster.open-cluster-management.io/placement: ztp-placement
+    cluster.open-cluster-management.io/decisiongroupindex: 2
+status:
+  decisions:
+    - clusterName: cls011
+      reason: ''
+    ...
+---
+apiVersion: cluster.open-cluster-management.io/v1beta1
+kind: PlacementDecision
+metadata:
+  name: ztp-placement-decision-3
+  namespace: ztp-acm-ns
+  labels:
+    cluster.open-cluster-management.io/placement: ztp-placement
+    cluster.open-cluster-management.io/decisiongroupindex: 2
+status:
+  decisions:
+    - clusterName: cls011
+      reason: ''
+    ...
+---
+apiVersion: cluster.open-cluster-management.io/v1beta1
+kind: PlacementDecision
+metadata:
+  name: ztp-placement-decision-4
+  namespace: ztp-acm-ns
+  labels:
+    cluster.open-cluster-management.io/placement: ztp-placement
+    cluster.open-cluster-management.io/decisiongroupindex: 3
+status:
+  decisions:
+    - clusterName: cls151
+      reason: ''
+    ...
+---
+apiVersion: cluster.open-cluster-management.io/v1beta1
+kind: PlacementDecision
+metadata:
+  name: ztp-placement-decision-5
+  namespace: ztp-acm-ns
+  labels:
+    cluster.open-cluster-management.io/placement: ztp-placement
+    cluster.open-cluster-management.io/decisiongroupindex: 3
+status:
+  decisions:
+    - clusterName: cls151
+      reason: ''
+    ...
+```
+
+The Placement Helper [library](https://github.com/open-cluster-management-io/api/blob/main/cluster/v1beta1/helpers.go) should provide functions to facilitate retrieve the cluster groups.
+
+#### Special case handling
+
+* Clusters have been removed from decision group index 1 and decision group index 2 has been created. In order to not shift current selected clusters in the decision groups, placement controller will keep append clusters to decision group index 2 till it reach max number of clusters per decision group then append new cluster to decision group index 1 till it reach the max number of clusters per decision group.
+
+
+### How Workload applier APIs will benefits from using placementStrategy
+
+The OCM APIs such as ManifestWorkReplicaSet or Policy can define how-to roll out the workloads considering the Decision Groups defined in the placement.
+For example; the below placementStrategy struct define the workload rollout type (All or Progressive) and decision groups to apply the workload first.
+
+```go
+type PlacementStrategy struct {
+	// Rollout type either All or Progressive.
+	// +kubebuilder:validation:Enum=All;Progressive
+	// +kubebuilder:default:=All
+	// +optional
+	RolloutType string `json:"rolloutType"`
+
+	// List of the decision groups labels to apply the workload first.
+	// +optional
+	DecisionGroupsToApplyFirst []string `json:"decisionGroupsToApplyFirst"`
```

Comment:
what does this mean? Is this for canary?

Reply:
Any decisionGroup defined under the placement->decisionStrategy, its label can be set here to apply the workloads for it first. Yes, Canary is one case any other decisionGroup label can be used.

Reply:
I think it depends on the consumers of the placement. I do not think we can define clearly the semantic of ApplyFirst in the placement API.

Reply:
The `DecisionGroupsToApplyFirst` is not part of placement API.  It is an example of how consumer API such as ManifestWorkReplicaSet can consume the DecisionStrategy.

Reply:
oh I see, that makes sense